GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -525,17 +528,22 @@ private void OnNewFileCreated(string relativePath, bool isDirectory)
                 {
                     if (isDirectory)
                     {
-                        GitCommandLineParser gitCommand = new GitCommandLineParser(this.Context.Repository.GVFSLock.GetLockedGitCommand());
-                        if (gitCommand.IsValidGitCommand)
-                        {
-                            // TODO(Mac): Ensure that when git creates a folder all files\folders within that folder are written to disk
+                        string lockedGitCommand = this.Context.Repository.GVFSLock.GetLockedGitCommand();
+                        GitCommandLineParser gitCommand = new GitCommandLineParser(lockedGitCommand);
+                        if (gitCommand.IsValidGitCommand)
+                        {
                             EventMetadata metadata = this.CreateEventMetadata(relativePath);
-                            metadata.Add("isDirectory", isDirectory);
-                            this.Context.Tracer.RelatedWarning(metadata, $"{nameof(this.OnNewFileCreated)}: Git created a folder, currently an unsupported scenario on Mac");
-                        }
-                        else
+                            metadata.Add(nameof(lockedGitCommand), lockedGitCommand);
+                            metadata.Add(TracingConstants.MessageKey.InfoMessage, "Git command created new folder");
+                            this.Context.Tracer.RelatedEvent(EventLevel.Informational, $"{nameof(this.OnNewFileCreated)}_GitCreatedFolder", metadata);
+
+                            // Record this folder as expanded so that GitIndexProjection will re-expand the folder

```

Comment:
> Some comments about the why would be useful here.

Sure will do

> E.g. we're doing this because a new file was created by a git command, and it's inside a folder that may need to project other items after the git command is completed. (Did I get that right?)

You did get that right.  The issue is that when git creates a folder it will only create files in that folder that are in ModifiedPaths.dat.  That's an issue because there could be other files in the folder (that are not in ModifiedPaths.dat) that VFS needs to create placeholders for.